pkg/errhandler/errhandler.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018-2020 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package errhandler
```

Comment:
Nice abstraction! however, I think the chose of name is incorrect, and the base principles. While an `errhandler` package would allow for abstracting error handling, this package deals more with `OCS` messages.

On my mind working with interfaces is more appropriate as a generalized idea of error handling, then, every package that deals with the ocs api would implement the `errhandler` interface. Otherwise if you want to keep this generic, more functions will be added for different packages.

Reply:
Hi @refs, thanks for the review! The main idea behind moving this to pkg was to prevent implementing it again. We're currently implementing the OCM HTTP service and it would have the same error codes as these, and I imagine all HTTP services would be having these, and the fact that we could pass messages to add details makes these quite generic. But we could definitely rename the package name to signify that it handles HTTP error codes.

Please let me know if that would be fine. And any suggestions that you have for the package name :smile:

Reply:
@ishank011 are really the OCS error responses the same for OCM? The ones that I've found in OCM only have a "MESSAGE" field:

```
{
  "message": "RESOURCE_NOT_FOUND"
}
```

Reply:
@labkode OCM defines it's own error codes and a few associated methods in https://github.com/cernbox/ocmd/blob/master/api/api.go#L14 which match with the OCS erros (https://github.com/cs3org/reva/blob/master/internal/http/services/owncloud/ocs/reqres.go#L54). We can add more if required. So instead of having separate implementations in both the services, this way we can just use a common package.

So what was previously done as https://github.com/cernbox/ocmd/blob/master/handlers/handlers.go#L308
```
ae := api.NewAPIError(api.APIErrorInvalidParameter).WithMessage("body is not json")
w.Write(ae.JSON())
```
can now be done in a similar way as OCS does.
```
errhandler.WriteError(w, r, MetaBadRequest.StatusCode, "body is not json"", nil)
```